docs/tekton-bundle-contracts.md

```diff
@@ -0,0 +1,38 @@
+<!--
+---
+linkTitle: "Tekton Bundle Contracts"
+weight: 8
+---
+-->
+# Tekton Bundle Contract
+
+When using a Tekton Bundle in a task or pipeline reference, the OCI artifact backing the
+bundle must adhere to the following contract.
+
+## Contract
+
+Only Tekton CRDs (eg, `Task` or `Pipeline`) may reside in a Tekton Bundle used as a Tekton
+bundle reference.
+
+Each layer of the image must map 1:1 with a single Tekton resource.
+
+Each layer must contain the following annotations:
+
+- `org.opencontainers.image.title` => `ObjectMeta.Name` of the resource
+- `cdf.tekton.image.kind` => `TypeMeta.Kind` of the resource, all lowercased (eg, `task`)
+- `cdf.tekton.image.apiVersion` => `TypeMeta.APIVersion` of the resource (eg 
+"tekton.dev/v1alpha1")  
+
+Each { `apiVersion`, `kind`, `title` } must be unique in the image. No resources of the
```

Comment:
This relates to the content (yaml/json inside a layer) or the annotations ? If it relates to the annotations, we may want to use the full name of it (e.g. `apiVersion` -> `cdf.tekton.image.apiVersion`, …)

Reply:
I will reword this to be more clear. The intention is that the combination of annotations on that layer are unique within the image so I will make it clear what fields we are referring to.